activejob/lib/active_job/core.rb

```diff
@@ -80,6 +80,7 @@ def serialize
       {
         "job_class"  => self.class.name,
         "job_id"     => job_id,
+        "provider_job_id" => provider_job_id,
```

Comment:
@jvanbaarsen The first assertion tests this, but perhaps it can be made clearer. I'll patch.